TimeTracking/pages/config_update.php

```diff
@@ -21,21 +21,27 @@
    http://www.mantisbt.org/forums/viewtopic.php?f=4&t=589	
 */
 
-form_security_validate( 'plugin_TimeTracking_config_update' );
+$fid = 'plugin_TimeTracking_config_update';
+form_security_validate( $fid );
 
 access_ensure_global_level( config_get( 'manage_plugin_threshold' ) );
 
-function maybe_set_option( $name, $value ) {
-	if ( $value != plugin_config_get( $name ) ) {
-		plugin_config_set( $name, $value );
-	}
-}
 
-maybe_set_option( 'edit_threshold', gpc_get_int( 'edit_threshold' ) );
-maybe_set_option( 'view_threshold', gpc_get_int( 'view_threshold' ) );
-maybe_set_option( 'reporting_threshold', gpc_get_int( 'reporting_threshold' ) );
-maybe_set_option( 'categories', gpc_get_string( 'categories' ) );
-maybe_set_option( 'stopwatch_enabled', gpc_get_int( 'stopwatch_enabled', OFF ) );
+$config = \TimeTrackingPlugin::getConfig();
+foreach($config as $opt => $default ) {
```

Comment:
Is this really needed? seems a lot of complexity for a bunch of options
What happens if at some point there are some config options that are not managed through this form?

Reply:
I personally do not like a lot seeing lot of similar lines that seems copy & paste, that's why I prefer what I consider a simple algorithm.
If I'm not wrong it is supposed that all config options will be managed by config page.